### ISSUE TITLE
Performance of Pkg.update

### ISSUE THREAD

**Issue by DhruvaSambrani** [JuliaLang/Pkg.jl#2348]
Running `] update` is extremely slow on Windows, and gets stuck at 
```
Updating registry at `C:?Users/username/.julia/registries/General` 
```
for well above 3 minutes. The rest of the steps are fine

Running `] update` on julia on WSL on the same device updates packages instantaneously.
Issue persists even if RealTime Scanning is stopped on Windows McAfee

**Comment by StefanKarpinski** [comment]
What version of Julia are you using?

**Comment by DhruvaSambrani** [comment]
V1.5.3 on both, WSL and Windows

**Comment by ianshmean** [comment]
I profiled `Pkg.update()` on a fresh depot in Windows on 1.5.3. It took 3m42s. Here's the profile report
https://gist.github.com/ianshmean/e31a31d7287c9a5b843d2dac7b56b2b2

It seems like a majority of time is spent during `cp`. Master uses `mv`.

I tried the same for nightly, which took 2m20s
https://gist.github.com/ianshmean/ccfe5ff232b87f468183507f651d1ebe

I also tried a 2nd time without wiping the depot and hit the error below, which highlighted that while installing a fresh registry no longer uses `cp` updating an existing one still does
https://github.com/JuliaLang/Pkg.jl/blob/fe759b2449b3ace291b6a5e43bd7ecff1767ff91/src/Registry/Registry.jl#L305

Could that be a `mv`?

```
julia> import Pkg; Pkg.update()
    Updating registry at `C:\Users\ian\.julia\registries\General`
ERROR: IOError: stat("C:\\Users\\ian\\.julia\\registries\\General"): permission denied (EACCES)
Stacktrace:
  [1] uv_error
    @ .\libuv.jl:97 [inlined]
  [2] stat(path::String)
    @ Base.Filesystem .\stat.jl:69
  [3] ispath
    @ .\stat.jl:311 [inlined]
  [4] checkfor_mv_cp_cptree(src::String, dst::String, txt::String; force::Bool)
    @ Base.Filesystem .\file.jl:301
  [5] cptree(src::String, dst::String; force::Bool, follow_symlinks::Bool)
    @ Base.Filesystem .\file.jl:324
  [6] cp(src::String, dst::String; force::Bool, follow_symlinks::Bool)
    @ Base.Filesystem .\file.jl:356
  [7] (::Pkg.Registry.var"#45#48"{String, Pkg.Registry.RegistryInstance, Base.SHA1})(tmp::String)
    @ Pkg.Registry C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.7\Pkg\src\Registry\Registry.jl:305
  [8] mktempdir(fn::Pkg.Registry.var"#45#48"{String, Pkg.Registry.RegistryInstance, Base.SHA1}, parent::String; prefix::String)
    @ Base.Filesystem .\file.jl:732
  [9] mktempdir
    @ .\file.jl:730 [inlined]
 [10] update(regs::Vector{Pkg.Registry.RegistrySpec}; io::Base.TTY, force::Bool)
    @ Pkg.Registry C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.7\Pkg\src\Registry\Registry.jl:296
 [11] up(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec}; level::Pkg.Types.UpgradeLevel, mode::Pkg.Types.PackageMode, update_registry::Bool, kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Pkg.API C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.7\Pkg\src\API.jl:306
 [12] up
    @ C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.7\Pkg\src\API.jl:300 [inlined]
 [13] up(pkgs::Vector{Pkg.Types.PackageSpec}; kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Pkg.API C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.7\Pkg\src\API.jl:143
 [14] up
    @ C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.7\Pkg\src\API.jl:141 [inlined]
 [15] up(; name::Nothing, uuid::Nothing, version::Nothing, url::Nothing, rev::Nothing, path::Nothing, mode::Pkg.Types.PackageMode, subdir::Nothing, kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Pkg.API C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.7\Pkg\src\API.jl:159
 [16] up()
    @ Pkg.API C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.7\Pkg\src\API.jl:151
 [17] top-level scope
    @ REPL[2]:1
```

**Comment by 0x0ade** [comment]
Have there been any updates regarding the "read registry directly from tar" idea that was brought up a while ago on a related issue? https://github.com/JuliaLang/Pkg.jl/issues/2014#issuecomment-690789260

**Comment by ianshmean** [comment]
@0x0ade I can't answer your question, but I recommend anyone tracking this issue to try out the upcoming 1.6-rc1 and see if things have improved.
There should have been a pretty decent improvement from 1.5.3

**Comment by KristofferC** [comment]
Should be fixed in 1.7 with the new compressed registry format.
